### PR TITLE
Add ERC-223 to token standards.

### PR DESCRIPTION
**Pull request by Dexaran**
Add ERC-223 token standard to the standards list.

Original ERC-223 proposal: https://github.com/ethereum/eips/issues/223

## Description

[EIP-223](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-223.md) is now officially submitted, so I would highlight the importance of adding this token standard to the standards list.

It is worth noting that several standards (including those presented in the list) were developed on the basis of the ERC-223, including [ERC-777](https://eips.ethereum.org/EIPS/eip-777), [ERC-677: TransferAndCall standard](https://github.com/ethereum/EIPs/issues/677).

The transferring logic of [ERC-721](https://eips.ethereum.org/EIPS/eip-721) is completely based on ERC-223.

The addition of this standard will fill in the missing link between the ERC20 standard and standards that already use the ERC223 patterns as well as help properly understanding the interaction patterns for Ether-like tokens.

### CODE REVIEW

**Comment by wackerow** [comment]
Hey @Dexaran, thanks for dropping this PR! 🙏... could you do us a favor and close either this one or #9646, depending which one you want reviewed? Note you can always push additional changes to a PR branch that you've opened to prevent having to open a new one if there are any issues.

**Comment by Dexaran** [comment]
Hello

Should I reopen the PR? Or is something needs to be changed?

I repeat that gatsby just timed out without giving any errors so I don't know what was wrong.

**Comment by Dexaran** [comment]
Hello,

It is one month since I opened a Pull Request. I've fixed inconsistencies and the PR passed gatsby tests. 

Can I get any details about what is the status of this?

**Comment by nhsz** [comment]
> Hello,
> 
> It is one month since I opened a Pull Request. I've fixed inconsistencies and the PR passed gatsby tests.
> 
> Can I get any details about what is the status of this?

cc @wackerow

**Comment by Dexaran** [comment]
cc @nhsz @wackerow @samajammin @corwintines @pettinarip @jmcook1186 @minimalsm

File: src/content/developers/docs/standards/tokens/erc-223/index.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        if(isContract(_to)) {
+            IERC223Recipient(_to).tokenReceived(msg.sender, _value, _empty);
+        }
+        emit Transfer(msg.sender, _to, _value, _empty);
+        return true;
+    }
+}
+```
+
+Now we want another contract to accept deposits of `tokenA` assuming that tokenA is a ERC-223 token. The contract must accept only tokenA and reject any other tokens. When the contract receives tokenA it must emit a `Deposit()` event and increase the value of the internal `deposits` variable.
```
Comment (wackerow) [inline src/content/developers/docs/standards/tokens/erc-223/index.md:127]:
```suggestion
Now we want another contract to accept deposits of `tokenA` assuming that tokenA is an ERC-223 token. The contract must accept only tokenA and reject any other tokens. When the contract receives tokenA it must emit a `Deposit()` event and increase the value of the internal `deposits` variable.
```

File: src/content/developers/docs/standards/tokens/erc-223/index.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
+---
+title: ERC-223 Token Standard
+description:
```
[the remark this answers is not in the document]
Comment (Dexaran) [reply src/content/developers/docs/standards/tokens/erc-223/index.md:3]: Ok, why not

File: src/content/developers/docs/standards/index.md [hunk 3]
```diff
@@ -40,10 +40,11 @@ More detailed information on these different types and categories can be found i
 ### Token standards {#token-standards}
 
 - [ERC-20](/developers/docs/standards/tokens/erc-20/) - A standard interface for fungible (interchangeable) tokens, like voting tokens, staking tokens or virtual currencies.
-  - [ERC-1363](https://eips.ethereum.org/EIPS/eip-1363) - Defines a token interface for ERC-20 tokens that supports executing recipient code after transfer or transferFrom, or spender code after approve
+  - [ERC-1363](https://eips.ethereum.org/EIPS/eip-1363) - Defines a token interface for ERC-20 tokens that supports executing recipient code after transfer or transferFrom, or spender code after approve.
+- [ERC-223](/developers/docs/standards/tokens/erc-223/) - A fungible tokens standard that makes tokens behave identical to Ether and supports token transfers handling on the recipients side.
```
[the remark this answers is not in the document]
Comment (Dexaran) [reply src/content/developers/docs/standards/index.md:44]: Well, ERC-223 is designed to behave similar to Eth.

ERC-20:

1. Has two transferring methods: `transfer()` function and `approve() + transferFrom()` pattern
2. You must use `transfer()` to deliver your tokens to an address owned by a user
3. You must use `approve() + transferFrom()` pattern to deposit your tokens to a contract, if you will use `transfer()` function then your tokens will get stuck and you will lose them.
4. ERC-20 token NEVER notifies a recipient on the incoming transaction

ETH:
1. Has only one transferring method. Notifies recipient on the incoming transaction.
2. You can send ETH to any address to deposit it.
3. You can deposit ETH to the contract by sending it to the contract address. This transaction will be handled on the recipients side.
4. You can't lose your ETH due to improperly handled transaction unless the recipient accepted your deposit on purpose.

ERC-223:
1. Has only one transferring method `transfer(...)` function. Notifies recipient on the incoming transaction similar to ETH transfers.
2. You can deposit ERC-223 tokens to any address by simply sending them via `transfer(...)` function.
3.  You can deposit ERC-223 to the contract by sending it to the contract address via `transfer(...)` function similar to how you could send ERC-223 tokens to any other address. This transaction will be handled on the recipients side.
4. You can't lose your ERC-223 tokens due to improperly handled transaction unless the recipient accepted your deposit on purpose by implementing a permissive `tokenReceived()` function.

File: src/content/developers/docs/standards/tokens/erc-223/index.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+- **How can we make a deposit to this contract?** - We can simply call the `transfer(address,uint256)` or `transfer(address,uint256,bytes)` function of the ERC-223 token and tell it to transfer some tokens to the address of the `RecipientContract`. That's it.
+- **What will happen if we transfer a ERC-20 token to this contract?** - Well, ERC-20 standard supports two methods of transferring tokens: `transfer` function and `approve + transferFrom` pattern. This is not possible to make a deposit with `transferFrom` function as the `RecipientContract` does not have any functions that subsequently call `transferFrom`. If a ERC-20 token is sent with `transfer` function to the address of the `RecipientContract` then unfortunately the tokens will be transferred from the sender's address to the address of the `RecipientContract` but the transfer will not be recognized i.e. `Deposit()` event will not be fired and `deposits` value will not change. There is also no way to filter or prevent unwanted ERC-20 deposits that are made with the `transfer` function.
+- **What if we want to execute some function after the token deposit is completed?**
```
[the remark this answers is not in the document]
Comment (Dexaran) [reply src/content/developers/docs/standards/tokens/erc-223/index.md]: Done https://github.com/Dexaran/ethereum-org-website/commit/c6b2dac9447732af2a661262f69bf6cfcc5c4b1b

**Comment by Dexaran** [comment]
@wackerow The requested changes are implemented.

File: src/content/developers/docs/standards/tokens/erc-223/index.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+### How can we make a deposit to this contract?
+
+We can simply call the `transfer(address,uint256)` or `transfer(address,uint256,bytes)` function of the ERC-223 token and tell it to transfer some tokens to the address of the `RecipientContract`. That's it.
+
+### What will happen if we transfer an ERC-20 token to this contract?
+
+The ERC-20 standard supports two methods of transferring tokens: `transfer` function and `approve + transferFrom` pattern. This is not possible to make a deposit with `transferFrom` function as the `RecipientContract` does not have any functions that subsequently call `transferFrom`. If an ERC-20 token is sent with `transfer` function to the address of the `RecipientContract` then unfortunately the tokens will be transferred from the sender's address to the address of the `RecipientContract` but the transfer will not be recognized i.e. `Deposit()` event will not be fired and `deposits` value will not change. There is also no way to filter or prevent unwanted ERC-20 deposits that are made with the `transfer` function.
+
+### What if we want to execute some function after the token deposit is completed?
```
Comment (wackerow) [inline src/content/developers/docs/standards/tokens/erc-223/index.md:164]: Thanks @Dexaran, looks good! Would just add some custom header ids back in and we'll be good to go:

```suggestion
## Frequently asked questions {#faq}

### What will happen if we send some tokenB to the contract? {#sending-tokens}

The transaction will fail and the transfer of tokens will simply not happen. The tokens will be returned to the sender's address.

### How can we make a deposit to this contract? {#contract-deposits}

We can simply call the `transfer(address,uint256)` or `transfer(address,uint256,bytes)` function of the ERC-223 token and tell it to transfer some tokens to the address of the `RecipientContract`. That's it.

### What will happen if we transfer an ERC-20 token to this contract? {#erc-20-transfers}

The ERC-20 standard supports two methods of transferring tokens: `transfer` function and `approve + transferFrom` pattern. This is not possible to make a deposit with `transferFrom` function as the `RecipientContract` does not have any functions that subsequently call `transferFrom`. If an ERC-20 token is sent with `transfer` function to the address of the `RecipientContract` then unfortunately the tokens will be transferred from the sender's address to the address of the `RecipientContract` but the transfer will not be recognized i.e. `Deposit()` event will not be fired and `deposits` value will not change. There is also no way to filter or prevent unwanted ERC-20 deposits that are made with the `transfer` function.

### What if we want to execute some function after the token deposit is completed? {#function-execution}
```

Comment (Dexaran) [reply src/content/developers/docs/standards/tokens/erc-223/index.md:164]: Sure

**Comment by Dexaran** [comment]
@wackerow can we move further with this?

**Comment by Dexaran** [comment]
cc @nhsz @wackerow @samajammin @corwintines @pettinarip @jmcook1186 @minimalsm

**Review by corwintines — APPROVED** [review]

**Pull request merged**